routes/signup.js

```diff
@@ -0,0 +1,60 @@
+const http = require("http");
+const express = require("express")
+const bodyParser = require("body-parser")
+const bcrypt = require('bcrypt');
+const cookieParser = require("cookie-parser")
+
+const { Schema,
+    userSchema,
+    User,
+    connect,
+    sessionSchema,
+    Session,
+    postSchema,
+    Post } = require("../database.js");
+
+const router = express.Router();
+
+router.get("/", async (req, res) => {
+    try {
+        //check if there's a session. If not, render the loging main page.
+        if (!req.session) {
+            res.render('signup');
+            return;
+        }
```

Comment:
Similarly here with the direction of the if, you would check the unusual situations first.